javaslang/src/main/java/javaslang/collection/AbstractMap.java

```diff
@@ -134,14 +134,7 @@ public M filter(Predicate<? super Tuple2<K, V>> predicate) {
     @SuppressWarnings("unchecked")
     @Override
     public <C> Map<C, M> groupBy(Function<? super Tuple2<K, V>, ? extends C> classifier) {
-        Objects.requireNonNull(classifier, "classifier is null");
```

Comment:
It depends on other modifications in this PR.
If I put it as a separate PR, they will have to be merged in a given order, and many changes will be duplicated, in order for the PRs to make sense